backend/src/org/commcare/session/SessionFrame.java

```diff
@@ -95,12 +98,10 @@ public String getFrameId() {
      * This snapshot can be referenced later to compare the eventual state
      * of the frame to an earlier point
      */
-    public void captureSnapshot() {
-        synchronized (steps) {
```

Comment:
Not sure what this lock was doing but why was it necessary to remove it / is it definitely ok to do so?